src/wallet/spend.cpp

```diff
@@ -357,17 +357,43 @@ bool CWallet::AttemptSelection(const CAmount& nTargetValue, const CoinEligibilit
 {
     setCoinsRet.clear();
     nValueRet = 0;
+    // Vector of results for use with waste calculation
+    // In order: calculated waste, selected inputs, selected value
+    // TODO: Use a struct representing the selection result
+    std::vector<std::tuple<CAmount, std::set<CInputCoin>, CAmount>> results;
```

Comment:
You may want to be a bit more explicit here whether the third value is the selected value, the selected effective value, or the target value. (I assume it's the first, but we've been juggling all of these, being more precise wouldn't hurt.)

Reply:
Done

Reply:
Do you anticipate having more coin selection algorithms in future? Why bother with the vector etc. rather than just direct comparison between the two results?

Reply:
Yes, #17526 will be updated to add another result to this vector.